### DISCUSSION TITLE
Get key frame

### DISCUSSION THREAD

**Discussion by TRINHHOANGANH** [stella-cv/stella_vslam#444] [Q&A]
Hi !!!
## Question
How can I get key frame from key frame ID and save its to folder.

**Comment by ymd-stella** [comment]
Implemented at https://github.com/stella-cv/stella_vslam/tree/wip/output_keyframes.

**Reply by TRINHHOANGANH** [reply]
Can you show me your code that gets keyframes ?

**Reply by ymd-stella** [reply]
Read the code.

**Reply by ymd-stella** [reply]
- https://github.com/stella-cv/stella_vslam/commit/f409de1ac0e0055427573853f6836992f45215c6
- https://github.com/stella-cv/stella_vslam/commit/d9c6045717cf4dad2507187e8e0180dbea81326d

**Comment by meowmoewrainbow** [comment]
Hi @ymd-stella 
I've also tried this code and it produced these bugs when I run make command
<pre>[ 63%] <font color="#4E9A06">Building CXX object example/CMakeFiles/run_video_slam.dir/run_video_slam.cc.o</font>
[ 64%] <font color="#4E9A06"><b>Linking CXX executable ../run_video_slam</b></font>
/usr/bin/ld: CMakeFiles/run_video_slam.dir/run_video_slam.cc.o: in function `mono_tracking(std::shared_ptr&lt;stella_vslam::system&gt; const&amp;, std::shared_ptr&lt;stella_vslam::config&gt; const&amp;, std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; const&amp;, std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; const&amp;, unsigned int, unsigned int, bool, bool, bool, std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; const&amp;, std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; const&amp;, std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; const&amp;, double)&apos;:
run_video_slam.cc:(.text+0x11a7): undefined reference to `stella_vslam::system::get_keyframe_images() const&apos;
collect2: error: ld returned 1 exit status
make[2]: *** [example/CMakeFiles/run_video_slam.dir/build.make:136: run_video_slam] Error 1
make[1]: *** [CMakeFiles/Makefile2:753: example/CMakeFiles/run_video_slam.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
</pre>

**Reply by ymd-stella** [reply]
If you are using OpenCV 4.*, CI passes, so you must be doing something wrong.

https://github.com/stella-cv/stella_vslam/actions/runs/3852598853/jobs/6564867502#step:4:208

**Comment by TRINHHOANGANH** [comment]
I save keyframe images to folder and this code creates transforms.json file. You can save all keyframes and their relationship( I think their relationship is graph).
